### PR TITLE
add public jenkins url to comments

### PR DESCRIPTION
**Pull request by PengTian0**
Add url to jenkins in cloud to the comments which is posted by jenkins after testing for PR.

Looks like:

BUILD [on-core #874](http://147.178.202.18/job/on-core/874/) : UNSTABLE
- *\* BUILD [build-config-setup  #101225](http://147.178.202.18/job/Templates/job/build-config-setup/101225/): SUCCESS
- *\* BUILD [unit-tests  #9343](http://147.178.202.18/job/Templates/job/unit-tests/9343/): SUCCESS
- *\* BUILD [smoke-tests  #3405](http://147.178.202.18/job/Templates/job/smoke-test/3405/): UNSTABLE
- *\* BUILD [on-taskgraph #1525](http://147.178.202.18/job/on-taskgraph/1525/): SUCCESS
  - *\* BUILD [build-config-setup  #101227](http://147.178.202.18/job/Templates/job/build-config-setup/101227/): SUCCESS
  - *\* BUILD [unit-tests  #9344](http://147.178.202.18/job/Templates/job/unit-tests/9344/): SUCCESS

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000 @changev

File: post-result.py.in [hunk 1]
```diff
@@ -10,10 +10,12 @@ python post-result.py \
 https://api.github.com/repos/RackHD/on-core/issues/${PullRequestId}/comments \
 http://rackhdci.lss.emc.com \
 http://rackhdci.lss.emc.com/job/on-core/851/ \
-'#851'
+'#851' \
+http://147.178.202.18/
 '''
 
-with open('${HOME}/.ghtoken', 'r') as file:
+HOME="/home/onrack"
```
Comment (changev) [inline post-result.py.in]: This shouldn't be hardcoded and in RackHD jenkins slave HOME is "/home/jenkins".

File: post-result.py.in [hunk 2]
```diff
@@ -176,16 +181,17 @@ def post_comments_to_github(comments, github_pr_url, headers):
...
+    public_build_url = BUILD_URL.replace(JENKINS_URL, PUBLIC_JENKINS_URL)
     try:
         build_data = get_build_data(BUILD_URL)
         if build_data:
             build_name = build_data['fullDisplayName']
             build_result = build_data['result']
-            OUTPUT +=  "BUILD " + build_name + " : " + build_result + "\n"
+            OUTPUT +=  "BUILD [" + build_name + "](" + public_build_url + ") : " + build_result + "\n"
 
             sub_build_outputs = get_sub_builds(BUILD_URL, 1)
```
Comment (changev) [inline post-result.py.in]: A tip that I don't know whether reasonable:

```
def get_sub_builds(builds_url, depth=1)
#when invoked here, just 
get_sub_builds(BUILD_URL)
```

get_sub_builds(BUILD_URL) is clear than get_sub_builds(BUILD_URL, 1)

File: post-result.py.in (hunk 1 above)
Comment (PengTian0) [reply post-result.py.in]: ah, it is used to test, I will remove that

File: post-result.py.in (hunk 2 above)
Comment (PengTian0) [reply post-result.py.in]: Thanks!
That's a good advice.

**Comment by PengTian0** [comment]
test this please

File: post-result.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 https://api.github.com/repos/RackHD/on-core/issues/${PullRequestId}/comments \
 http://rackhdci.lss.emc.com \
 http://rackhdci.lss.emc.com/job/on-core/851/ \
-'#851'
+'#851' \
+http://147.178.202.18/
```
Comment (panpan0000) [inline post-result.py]: where do you document the meaning of each parameter?

File: post-result.py [hunk 4]
```diff
@@ -22,6 +23,7 @@
 JENKINS_URL = sys.argv[2]
 BUILD_URL = sys.argv[3]
 BUILD_NAME = sys.argv[4]
+PUBLIC_JENKINS_URL = sys.argv[5]
```
Comment (panpan0000) [inline post-result.py]: Don't you check the argc ? I meant , are those 5 parameter are all optional ?  if not, should we return error in script if no all 5 parameter given ?

**Comment by panpan0000** [comment]
@anhou @yyscamper  , can you help to review and approve this small PR?
so that @PengTian0  can demo this in global sprint review meeting

File: post-result.py (hunk 4 above)
Comment (PengTian0) [reply post-result.py]: The 5 parameters are required.
I will add a function to parse all the arguments which will also add comments and check arguments

File: post-result.py (hunk 3 above)
Comment (PengTian0) [reply post-result.py]: I will add a function to parse arguments which include "help" for each argument.

**Comment by changev** [comment]
test this please

**Review by yyscamper — APPROVED** [review]

**Comment by PengTian0** [comment]
test this please

**Comment by panpan0000** [comment]
:+1

**Pull request merged**